### PR TITLE
Converting HASS Configurator to the new configuration template format.

### PR DESCRIPTION
**Pull request by tmcarr**
**Description:**

Converted the hass.io configurator page to the new config format. As per this request:

https://github.com/home-assistant/home-assistant.io/issues/6385

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Review by WofWca — CHANGES_REQUESTED** [review]
Why are `sesame` variables missing?

**Comment by tmcarr** [comment]
@WofWca There was no description of them in the first place. Both the code example, and the config listing were out of sync. I did the best I could given what was there, but I'll totally add more if anyone has more context on them.

**Comment by WofWca** [comment]
What do you mean by "there was no description"?
>- **sesame** (*Optional*): Secret token to dynamically allow access from the IP the request originates from. Open your bookmark https://hassio.yourdomain.com:8123/somesecretnobodycanguess while `allowed_networks` is set to `[]` and your IP will get whitelisted. You can use the _Network status_ menu to revoke IP addresses for which access has been granted. Regular authentication is still required.
> - **sesame_totp_secret** (*Optional*): Like the `sesame` option, but instead as Base32 encoded secret string must be provided. This string then can be added to a TOTP App like Google Authenticator. This way you get a 6-digit `sesame` that changes every 30 seconds.

By "out of sync" you mean that they're missing from the code example? It's supposed to be that way, because they're optional.

**Comment by tmcarr** [comment]
@WofWca Gotcha, added them in.

File: source/_addons/configurator.markdown [hunk 1]
```diff
@@ -62,22 +62,72 @@ Screenshot of the HASS Configurator.
...
+ignore_ssl:
+  description: Ignore SSL errors when accessing the Home Assistant API.
+  required: true
+  type: string
+loglevel:
+  description: You can change the logging level from the default value info if you want to. Valid values are debug, info, warning, error, critical.
+  required: false
+  type: string
+sesame:
+  description: Secret token to dynamically allow access from the IP the request originates from. Open your bookmark https://hassio.yourdomain.com:8123/somesecretnobodycanguess while allowed_networks is set to [] and your IP will get whitelisted. You can use the Network status menu to revoke IP addresses for which access has been granted. Regular authentication is still required.
```
Comment (DubhAd) [inline source/_addons/configurator.markdown]: Please use markup as per the original

File: source/_addons/configurator.markdown [hunk 2]
```diff
@@ -62,22 +62,72 @@ Screenshot of the HASS Configurator.
...
+  required: false
+  type: string
+sesame_totp_secret:
+  description: Like the sesame option, but instead as Base32 encoded secret string must be provided. This string then can be added to a TOTP App like Google Authenticator. This way you get a 6-digit sesame that changes every 30 seconds.
```
Comment (DubhAd) [inline source/_addons/configurator.markdown]: Please use markup as per the original

File: source/_addons/configurator.markdown [hunk 3]
```diff
@@ -62,22 +62,72 @@ Screenshot of the HASS Configurator.
...
+  type: int
+ignore_pattern:
+  description: Regex of files and folders to ignore in the UI
+  required: true
+  type: string
+dirsfirst:
+  description: List directories before files in the file browser.
+  required: true
+  type: string
+enforce_basepath:
```
Comment (DubhAd) [inline source/_addons/configurator.markdown:108]: Please use markup as per the original

File: source/_addons/configurator.markdown [hunk 4]
```diff
@@ -62,22 +62,72 @@ Screenshot of the HASS Configurator.
...
+allowed_networks:
+  description: Limit access to the configurator by adding allowed IP addresses/networks to the list.
+  required: true
+  type: string
+banned_ips:
+  description: List of statically banned IP addresses.
+  required: true
+  type: string
+banlimit:
+  description: Ban access from IPs after banlimit failed login attempts. The default value 0 disables this feature. Restart the add-on to clear the list of banned IP addresses.
```
Comment (DubhAd) [inline source/_addons/configurator.markdown]: Please use markup as per the original

File: source/_addons/configurator.markdown [hunk 5]
```diff
@@ -62,22 +62,72 @@ Screenshot of the HASS Configurator.
...
+password:
+  description: Set a password for access.
+  required: true
+  type: string
+ssl:
+  description: Enable or Disable SSL/TLS for the editor.
+  required: true
+  type: string
+certfile:
+  description: Set the path the your SSL certificate if the ssl-option is set to true.
```
Comment (DubhAd) [inline source/_addons/configurator.markdown]: Please use markup as per the original

File: source/_addons/configurator.markdown [hunk 6]
```diff
@@ -62,22 +62,72 @@ Screenshot of the HASS Configurator.
...
+  required: true
+  type: string
+keyfile:
+  description: Set the path the your SSL private key if the ssl-option is set to true.
```
Comment (DubhAd) [inline source/_addons/configurator.markdown]: Please use markup as per the original

File: source/_addons/configurator.markdown [hunk 7]
```diff
@@ -62,22 +62,72 @@ Screenshot of the HASS Configurator.
...
+notify_service:
+  description: Specify a custom notify-service to be used to push notifications.
+  required: true
+  type: string
+ignore_ssl:
+  description: Ignore SSL errors when accessing the Home Assistant API.
+  required: true
+  type: string
+loglevel:
+  description: You can change the logging level from the default value info if you want to. Valid values are debug, info, warning, error, critical.
```
Comment (DubhAd) [inline source/_addons/configurator.markdown]: Please use markup as per the original

**Review by DubhAd — CHANGES_REQUESTED** [review]

File: source/_addons/configurator.markdown (hunk 5 above)
Comment (tmcarr) [reply source/_addons/configurator.markdown]: This *IS* what was written in the original....

```
- **certfile** (*Required*): Set the path the your SSL certificate if the ssl-option is set to `true`.
```

Comment (tmcarr) [reply source/_addons/configurator.markdown]: I see what you mean, fix submitted.

**Comment by tmcarr** [comment]
@DubhAd Fixed.

File: source/_addons/configurator.markdown [hunk 8]
```diff
@@ -62,22 +62,72 @@ Screenshot of the HASS Configurator.
...
+  type: string
+allowed_networks:
+  description: Limit access to the configurator by adding allowed IP addresses/networks to the list.
+  required: true
+  type: string
+banned_ips:
+  description: List of statically banned IP addresses.
+  required: true
+  type: string
+banlimit:
```
Comment (fabaff) [inline source/_addons/configurator.markdown:94]: Default is missing.

Comment (tmcarr) [reply source/_addons/configurator.markdown:94]: Fixed.

**Comment by tmcarr** [comment]
I can't tell whats left here... It says there are open change requests, but I cant find them?

**Comment by tmcarr** [comment]
@DubhAd Any ideas what I missed here? I think I covered everything... I can't figure out why it still shows you as having requested changes.

File: source/_addons/configurator.markdown [hunk 9]
```diff
@@ -62,22 +62,73 @@ Screenshot of the HASS Configurator.
...
+ignore_ssl:
+  description: Ignore SSL errors when accessing the Home Assistant API.
+  required: true
+  type: string
+loglevel:
+  description: You can change the logging level from the default value info if you want to. Valid values are `debug`, `info`, `warning`, `error`, `critical`.
+  required: false
+  type: string
+sesame:
+  description: Secret token to dynamically allow access from the IP the request originates from. Open your bookmark https://hassio.yourdomain.com:8123/somesecretnobodycanguess while `allowed_networks` is set to [] and your IP will get whitelisted. You can use the Network status menu to revoke IP addresses for which access has been granted. Regular authentication is still required.
```
Comment (tjorim) [inline source/_addons/configurator.markdown]: Put `[]` back between code quotes.

File: source/_addons/configurator.markdown [hunk 10]
```diff
@@ -62,22 +62,73 @@ Screenshot of the HASS Configurator.
...
+  type: string
+notify_service:
+  description: Specify a custom notify-service to be used to push notifications.
+  required: true
+  type: string
+ignore_ssl:
+  description: Ignore SSL errors when accessing the Home Assistant API.
+  required: true
+  type: string
+loglevel:
```
Comment (tjorim) [inline source/_addons/configurator.markdown:117]: Add `default: info`
(update the description a bit as well please, it doesn't need to mention 'default value info')

File: source/_addons/configurator.markdown [hunk 11]
```diff
@@ -62,22 +62,73 @@ Screenshot of the HASS Configurator.
...
+  required: true
+  type: string
+password:
+  description: Set a password for access.
+  required: true
+  type: string
+ssl:
+  description: Enable or Disable SSL/TLS for the editor.
+  required: true
+  type: string
```
Comment (tjorim) [inline source/_addons/configurator.markdown]: ssl-type is boolean, default: false

File: source/_addons/configurator.markdown [hunk 12]
```diff
@@ -62,22 +62,73 @@ Screenshot of the HASS Configurator.
...
+  type: int
+  default: 0
+ignore_pattern:
+  description: Regex of files and folders to ignore in the UI
+  required: true
+  type: string
+dirsfirst:
+  description: List directories before files in the file browser.
+  required: true
+  type: string
```
Comment (tjorim) [inline source/_addons/configurator.markdown]: dirsfirst-type is boolean, default: false

File: source/_addons/configurator.markdown [hunk 13]
```diff
@@ -62,22 +62,73 @@ Screenshot of the HASS Configurator.
...
+enforce_basepath:
+  description: If set to `true`, access is limited to files within the `/config` directory.
+  required: true
+  type: string
```
Comment (tjorim) [inline source/_addons/configurator.markdown]: type is boolean, default: false

File: source/_addons/configurator.markdown [hunk 14]
```diff
@@ -62,22 +62,73 @@ Screenshot of the HASS Configurator.
...
+allowed_networks:
+  description: Limit access to the configurator by adding allowed IP addresses/networks to the list.
+  required: true
+  type: string
+banned_ips:
+  description: List of statically banned IP addresses.
+  required: true
+  type: string
+banlimit:
+  description: Ban access from IPs after `banlimit` failed login attempts. This feature is disabled by default. Restart the add-on to clear the list of banned IP addresses.
```
Comment (tjorim) [inline source/_addons/configurator.markdown]: I would suggest changing ". This feature is disabled by default." a bit,
what about ", setting the value to 0 disables this feature."?

File: source/_addons/configurator.markdown [hunk 15]
```diff
@@ -62,22 +62,73 @@ Screenshot of the HASS Configurator.
...
+  type: string
+enforce_basepath:
+  description: If set to `true`, access is limited to files within the `/config` directory.
+  required: true
+  type: string
+notify_service:
+  description: Specify a custom notify-service to be used to push notifications.
+  required: true
+  type: string
+ignore_ssl:
```
Comment (tjorim) [inline source/_addons/configurator.markdown]: `ignore_ssl` isn't used anymore. Please remove it from the example configuration as well.

**Review by tjorim — CHANGES_REQUESTED** [review]
Thanks for updating this, I left some feedback.

File: source/_addons/configurator.markdown (hunk 10 above)
Comment (tmcarr) [reply source/_addons/configurator.markdown:117]: Whoops, fixing that.

File: source/_addons/configurator.markdown (hunk 11 above)
Comment (tmcarr) [reply source/_addons/configurator.markdown]: I dont think there IS a boolean type for this. In the docs I read, true/false fell into string. Let me look again.

Comment (tmcarr) [reply source/_addons/configurator.markdown]: Annnnnnnnnd brainfart. Fixing.

**Review by DubhAd — DISMISSED** [review]

File: source/_addons/configurator.markdown [hunk 16]
```diff
@@ -57,27 +57,77 @@ Screenshot of the HASS Configurator.
...
+  required: true
+  type: string
+banned_ips:
+  description: List of statically banned IP addresses.
+  required: true
+  type: string
+banlimit:
+  description: Ban access from IPs after `banlimit` failed login attempts, setting the value to 0 disables this feature. Restart the add-on to clear the list of banned IP addresses.
+  required: true
+  type: int
```
Comment (frenck) [inline source/_addons/configurator.markdown]:
```suggestion
  type: integer
```

File: source/_addons/configurator.markdown [hunk 17]
```diff
@@ -57,27 +57,77 @@ Screenshot of the HASS Configurator.
...
+  default: 0
+ignore_pattern:
+  description: Regex of files and folders to ignore in the UI
```
Comment (frenck) [inline source/_addons/configurator.markdown]:
```suggestion
  description: Regex of files and folders to ignore in the UI.
```

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by tmcarr** [comment]
@frenck Seems good now?

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
@tmcarr LGTM! 👍 

Happy Hacktoberfest 🎉

**Pull request merged**